### ISSUE TITLE
Forcefield from AmberPrmtopfile - Question

### ISSUE THREAD

**Issue by pguillem** [openmm/openmm#3877]
Hello everyone,

- I have loaded prmtop and inpcrd files from Amber, which creates an openmm topology and system.
- The system contains non-standard residues and small molecules, defined in other amber files. 
- I have simulated the system in openmm and it works perfectly.

**Quick questions**

1) Can I obtain a Forcefield object from the System created from the PRMTOP?. Could I mutate a standard residue in the topology and recreate the system without having to load the non-standard definitions into openmm?. (I couldn't find a way to parse amber .frcmod and .off files into openmm .xml definitions, this is why we use tleap to build the input files instead of the usual PDBFile+Forcefield to create the System).

2) I want to mutate **standard** residues in the original topology from amber and run in openMM. If I do so, would I have to create a new System with that topology, and apply a forcefield from scratch?. This would imply loading the non-standard definitions into openmm, which is what I'm trying to avoid.

In other words... since the mutation I need is a single residue, I want to reuse as much as possible from the existing System object that I imported from amber. My third choice is to invoke tleap from python to recreate the prmtop/inpcrd files after mutating the topology. But that's far from ideal.

Any ideas?

Thanks in advance.
Pedro

**Comment by peastman** [comment]
It isn't possible in the way you discuss.  Let me describe the information found in the different objects.

- A Topology contains chemical information: a set of atoms connected by bonds.
- A ForceField contains rules for assigning parameters to a Topology.  "When two atoms of these types are bonded, model the bond as a harmonic force with this spring constant and this equilibrium length."
- A System contains the particular parameters that have been assigned to a Topology.  It doesn't have any information about how those parameters were chosen.
- A prmtop file is similar to a System.  It contains parameters for a particular system, but no information about how they were assigned.

That's why you can't go backward and create a ForceField from a System.  It doesn't know what rules were used to assign the parameters.  And if you mutated a residue in the System, it wouldn't know how to assign parameters to the new residue.  The workflow always has to be Topology + ForceField -> System.  AmberTools works similarly, with different programs and files of course, but the same basic flow of information.

**Comment by aizvorski** [comment]
@pguillem I would just add one thing to what Peter wrote: it is in principle possible to take a Topology+System and make a ForceField (or forcefield xml file) with parameters for each unique residue in the topology (at least assuming that each occurence of the same residue has the same parameters).  At that point you should be able to use that forcefield to parametrize any other topology which has only copies of those residues.  The code to do that doesn't exist yet, but it would be a nice capability to add.

Probably the trickiest part of that is the bonds between residues: those would have atoms involved which come from different residues.  If you're somewhat lucky (dealing with something like a standard protein forcefield) the parameters involving those bonds/atoms would be all the same, and so not dependend on each residue's neighbors.

**Comment by pguillem** [comment]
Dear @peastman and @aizvorski,

Thank you for investing your valuable time in explaining the details. It helped a lot in understanding each object and how to address our particular use case. We ended up circumventing the problem using tleap.

I also believe that some sort of "parameter deconvolution" into a forcefield would be interesting (academically speaking), but considering what you both explained and the fact arbitrary combinations of different force terms could yield similar outputs makes it extremely tricky (or at most, unreliable?). I can't begin to think of a reverse transform that would approximate such a thing accurately.

Wishing the best energy for 2023!.

Best
Pedro
